uefi/src/proto/shell_params.rs

```diff
@@ -0,0 +1,45 @@
+//! `ShellParams` protocol
+
+use crate::proto::unsafe_protocol;
+use crate::{data_types, Char16};
+use core::slice::from_raw_parts;
+use uefi_raw::protocol::shell_params::ShellParametersProtocol;
+
+#[cfg(feature = "alloc")]
+use crate::CStr16;
+#[cfg(feature = "alloc")]
+use alloc::string::String;
+#[cfg(feature = "alloc")]
+use alloc::string::ToString;
+
+/// The ShellParameters protocol.
+#[derive(Debug)]
+#[repr(transparent)]
+#[unsafe_protocol(ShellParametersProtocol::GUID)]
+pub struct ShellParameters(ShellParametersProtocol);
+
+impl ShellParameters {
+    /// Get an iterator of the shell parameter arguments
+    #[cfg(feature = "alloc")]
+    pub fn get_args(&self) -> impl Iterator<Item = String> {
+        unsafe {
+            from_raw_parts(
+                self.0.argv.cast::<*const data_types::chars::Char16>(),
+                self.0.argc,
+            )
+            .iter()
+            .map(|x| CStr16::from_ptr(*x).to_string())
+        }
+    }
+
+    /// Get a slice of the args, as Char16 pointers
+    #[must_use]
+    pub fn get_args_slice(&self) -> &[*const Char16] {
```

Comment:
Along with changing the other method to be available without `alloc`, I think we can drop the `pub` from this method and keep it as an internal helper.